### PR TITLE
samples: nrf9160: modem_shell: add MQTT support for A-GPS and P-GPS

### PR DESCRIPTION
**Pull request by tokangas**
Added MQTT support for "gnss" command A-GPS and P-GPS.
Replaced separate GNSS work queue with modem_shell common work queue.
Cleaned up MQTT and REST related configuration flags.

### CODE REVIEW

**Review by peknis — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/gnss/gnss.c [hunk 1]
```diff
@@ -737,6 +742,7 @@ static void handle_timeout_work_fn(struct k_work *item)
 }
 
 #if defined(CONFIG_NRF_CLOUD_PGPS)
+#if !defined(CONFIG_NRF_CLOUD_MQTT)
```
Comment (jhirsi) [inline samples/nrf9160/modem_shell/src/gnss/gnss.c:745]: I was wondering where is the PGPS support?

File: samples/nrf9160/modem_shell/src/gnss/gnss.c [hunk 2]
```diff
@@ -39,14 +39,16 @@
...
+/* Verify that MQTT or REST is enabled */
+BUILD_ASSERT(IS_ENABLED(CONFIG_NRF_CLOUD_MQTT) || IS_ENABLED(CONFIG_NRF_CLOUD_REST),
+	     "CONFIG_NRF_CLOUD_MQTT or CONFIG_NRF_CLOUD_REST transport must be enabled");
 #endif
 
 #define GNSS_DATA_HANDLER_THREAD_STACK_SIZE 1536
 #define GNSS_DATA_HANDLER_THREAD_PRIORITY   5
 
-#define GNSS_WORKQ_THREAD_STACK_SIZE 2048
+#define GNSS_WORKQ_THREAD_STACK_SIZE 3584
```
[the remark this answers is not in the document]
Comment (tokangas) [reply samples/nrf9160/modem_shell/src/gnss/gnss.c]: Replaced GNSS work queue with the modem_shell common work queue as discussed offline.

File: samples/nrf9160/modem_shell/src/gnss/gnss.c (hunk 1 above)
Comment (tokangas) [reply samples/nrf9160/modem_shell/src/gnss/gnss.c:755]: It just magically works. 😄 The P-GPS implementation fetches predictions when `nrf_cloud_pgps_init()`, `nrf_cloud_pgps_preemptive_updates()` or `nrf_cloud_pgps_notify_prediction()` is called (the so called indirect method).

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/networking/nrf_cloud_pgps.html#requesting-and-processing-p-gps-data

**Comment by tokangas** [comment]
@trantanen @jhirsi I think I've handled all your comments.

The GNSS module now uses the common modem_shell work queue. I also restored the A-GPS filtered ephemerides implementation, but it can only be used with REST-only configuration.

**Review by jhirsi — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Review by hiltunent — APPROVED** [review]

**Pull request merged**